### PR TITLE
Appveyor script

### PR DESCRIPTION
**Pull request by fgregg**
This appveyor script will 

* built and test BTrees on windows for multiple versions of python and 32 and 64bit architectures
* on a tag, deploy window wheels to pypi

You will need to 
1. setup appveyor for the BTrees repo: https://www.appveyor.com/docs
2. create an encrypted variable for "password=the_zope.wheelbuilder_password" and let me know what it is, I'll incorporate it into the script. https://www.appveyor.com/docs/build-configuration#secure-variables

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - echo [distutils]                                  > %USERPROFILE%\\.pypirc
+  - echo index-servers =                             >> %USERPROFILE%\\.pypirc
+  - echo     pypi                                    >> %USERPROFILE%\\.pypirc
+  - echo [pypi]                                      >> %USERPROFILE%\\.pypirc
+  - echo repository=https://pypi.python.org/pypi     >> %USERPROFILE%\\.pypirc
+  - echo username=zope.wheelbuilder                  >> %USERPROFILE%\\.pypirc
+  - echo password=%password%                         >> %USERPROFILE%\\.pypirc
+  - set HOME=%USERPROFILE%
+  - pip install wheel twine
+  - ps: if($env:APPVEYOR_REPO_TAG -eq $TRUE) { python -W ignore setup.py && twine upload dist/* }
```
Comment (mgedmin) [inline appveyor.yml]: Shouldn't that `python setup.py` be `python setup.py bdist_wheel bdist_egg`?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - python : 33
+    - python : 33-x64    
+    - python : 34
+    - python : 34-x64
+    - python : 35	
+    - python : 35-x64
+
+install:
+  - "SET PATH=C:\\Python%PYTHON%;c:\\Python%PYTHON%\\scripts;%PATH%"
+  - echo "C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd" /x64 > "C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\bin\amd64/vcvars64.bat"
```
Comment (mgedmin) [inline appveyor.yml]: That slash in `/vcvarsall.bar` at the end looks strange after all the other paths used backslashes.

**Comment by mgedmin** [comment]
BTW the only other ZopeFoundation project that uses appveyor that I know of is zope.interface:
https://github.com/zopefoundation/zope.interface/blob/master/appveyor.yml

(It doesn't build wheels or upload them to PyPI -- at the moment.)

The approach to Python versions seems quite different than here.  Any reason for that?  Can we simplify the zope.interface appveyor.yml?

File: appveyor.yml (hunk 1 above)
Comment (fgregg) [inline appveyor.yml]: It looks like some body is already building window wheels for BTree? https://pypi.python.org/pypi/BTrees

File: appveyor.yml (hunk 2 above)
Comment (fgregg) [inline appveyor.yml]: I think this accomplishes the same thing as `DISTUTILS_USE_SDK: "1" 

I didn't know about that approach, which seems better.

File: appveyor.yml (hunk 1 above)
Comment (tseaver) [inline appveyor.yml]: > It looks like some body is already building window wheels for BTree?

That would be the [`zope.wineggbuilder`](https://github.com/zopefoundation/zope.wineggbuilder/) deployment, which is on the same hard-to-maintain machine as the [`zope.winbot`](https://github.com/zopefoundation/zope.winbot/) deployment.

Comment (mgedmin) [inline appveyor.yml]: Note how existing wineggbuilder uploads lack support for Python [3.3](https://github.com/zopefoundation/zope.wineggbuilder/issues/3), [3.4](https://github.com/zopefoundation/zope.wineggbuilder/issues/7) and [3.5](https://github.com/zopefoundation/zope.wineggbuilder/issues/8) (and also for [wheels](https://github.com/zopefoundation/zope.wineggbuilder/issues/2)).

We'll probably want to disable wineggbuilder for BTrees as soon as Appveyor starts working.  That can be done by specifying maxVersion in https://github.com/zopefoundation/zope.wineggbuilder/blob/master/rackspace.ini#L274

File: appveyor.yml [hunk 3]
```diff
@@ -35,6 +36,6 @@ deploy_script:
   - echo password=%password%                         >> %USERPROFILE%\\.pypirc
   - set HOME=%USERPROFILE%
   - pip install wheel twine
-  - ps: if($env:APPVEYOR_REPO_TAG -eq $TRUE) { python -W ignore setup.py && twine upload dist/* }
+  - ps: if($env:APPVEYOR_REPO_TAG -eq $TRUE) { python -W ignore setup.py bdist_wheel bdist_egg && twine upload dist/* }
```
Comment (mgedmin) [inline appveyor.yml]: I'd completely forgotten that the old wineggbuilder used to upload both .exe and .egg files.  Question for the audience: do we still need .exe installers?

My gut feeling is that people who still use zc.buildout will need .egg, and people who have migrated to pip will need .whl, but probably nobody needs .exe.  Unless there are still people who install stuff manually by downloading installers from PyPI and running them?

**Comment by fgregg** [comment]
Do you need anything else from me right now?

**Comment by mgedmin** [comment]
Here's the encrypted zope.wheelbuilder password for Appveyor: RtpeKCle25vCixaUcJBu6Q==

Here's the appveyor project page, under my personal appveyor account: https://ci.appveyor.com/project/mgedmin/btrees

How does ZopeFoundation intend to manage appveyor accounts?

**Comment by mgedmin** [comment]
I don't know how to ask AppVeyor to try-build this PR.

I expect it will try a build when you push a commit to update the encrypted secret.

**Comment by fgregg** [comment]
Hi @mgedmin it looks like you have to do a kind of workaround for this http://help.appveyor.com/discussions/problems/2209-node-grunt-build-specify-a-project-or-solution-file-the-directory-does-not-contain-a-project-or-solution-file#comment_36945393

**Comment by mgedmin** [comment]
Why do you say that?  The current appveyor failures are expected: none of the branches being built had an `appveyor.yml`.

Now we're waiting for you to push any commit (e.g. updating the encrypted secret) to this branch so Appveyor gets a chance.

**Comment by fgregg** [comment]
Woops! Sorry it's building. Appveyor disables deploys in Pull Requests, so that's why that section is no showing up.

**Comment by fgregg** [comment]
Okay, I think this good to merge.

**Comment by mgedmin** [comment]
Let's get the broken test fixed (or at least skip it on 64-bit windows with `@unittest.skipIf` or something) before merging this.

**Comment by fgregg** [comment]
testing the sys.maxsize here, since this is the only PR with appveyor tests

**Comment by tseaver** [comment]
Heh, @mgedmin just turned off appveyor for branches w/o `.appveyor.yml`.

**Comment by tseaver** [comment]
@fgregg should I close #34?

**Comment by fgregg** [comment]
@tseaver sure, we can get it working in this branch.

**Comment by fgregg** [comment]
Okay the sys.maxsize idea won't work because 

on windows python 3, x64

```
print(sys.maxsize) 
9223372036854775807
```

**Comment by tseaver** [comment]
For reference:

- `BTrees.tests.common.LARGEST_64_BITS` is 9223372036854775807

- `BTrees.tests.common.SMALLEST_64_BITS` is -9223372036854775808

and the code raising the OverflowError is *not* in BTrees C code:  when it detects a too-large / too-small value, it raises `ValueError`.

**Comment by fgregg** [comment]
hi @tseaver I merged in #35 and it's not a fix, yet.

**Comment by fgregg** [comment]
@mgedmin @tseaver maybe this PR should be merged into a branch, so that you can get the benefit of testing on appveyor without me being a bottleneck?

**Comment by tseaver** [comment]
@fgregg I merged / rebased the non-reverted commits from your branch onto the `fix-appveyor-build` branch in this repo:  see #36.

**Comment by tseaver** [comment]
@fgregg  Closing in favor of my changes on top of yours in #36.

**Pull request closed without merging**